tembo-operator/src/dedicated_networking.rs

```diff
@@ -0,0 +1,443 @@
+                .await
+                .map_err(|e| {
+                    error!("Failed to reconcile network policies: {:?}", e);
+                    e
+                })?;
+            debug!(
+                "Handling primary service ingress for CoreDB instance: {}",
+                cdb.name_any()
+            );
+            reconcile_dedicated_networking_service(
+                cdb,
+                client.clone(),
+                &ns,
+                dedicated_networking.public,
+                false,
+                &dedicated_networking.serviceType,
+                basedomain,
+            )
+            .await
+            .map_err(|e| {
+                error!("Failed to reconcile primary service ingress: {:?}", e);
+                e
+            })?;
+
+            if dedicated_networking.includeStandby {
+                debug!(
+                    "Handling standby service ingress for CoreDB instance: {}",
+                    cdb.name_any()
+                );
+                reconcile_dedicated_networking_service(
+                    cdb,
+                    client.clone(),
+                    &ns,
+                    dedicated_networking.public,
+                    true,
+                    &dedicated_networking.serviceType,
+                    basedomain,
+                )
+                .await
+                .map_err(|e| {
+                    error!("Failed to reconcile standby service ingress: {:?}", e);
+                    e
+                })?;
+            } else {
+                debug!(
+                    "Standby service is not included. Deleting standby service for CoreDB instance: {}",
+                    cdb.name_any()
+                );
+                delete_dedicated_networking_service(client.clone(), &ns, &cdb.name_any(), true)
+                    .await
+                    .map_err(|e| {
+                        error!("Failed to delete standby service: {:?}", e);
+                        e
+                    })?;
+            }
+        } else {
+            debug!(
+                "Dedicated networking is disabled. Deleting services and ingress routes for CoreDB instance: {}",
+                cdb.name_any()
+            );
+
+            delete_dedicated_networking_service(client.clone(), &ns, &cdb.name_any(), false)
+                .await
+                .map_err(|e| {
+                    error!("Failed to delete primary service: {:?}", e);
+                    e
+                })?;
+
+            delete_dedicated_networking_service(client.clone(), &ns, &cdb.name_any(), true)
+                .await
+                .map_err(|e| {
+                    error!("Failed to delete standby service: {:?}", e);
+                    e
+                })?;
+        }
+    } else {
+        debug!(
+            "Dedicated networking is not configured for CoreDB instance: {}",
+            cdb.name_any()
+        );
+    }
```

Comment:
In the case of dedicated networking not enabled, it should still clean up resources. For example, enable dedicated networking, then delete the configuration from coredb, it should clean up the resources